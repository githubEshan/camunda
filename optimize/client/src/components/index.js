/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
 * one or more contributor license agreements. See the NOTICE file distributed
 * with this work for additional information regarding copyright ownership.
 * Licensed under the Camunda License 1.0. You may not use this file
 * except in compliance with the Camunda License 1.0.
 */

export {Analysis} from './Analysis';
export {Header} from './Header';
export {PrivateRoute} from './PrivateRoute';
export {Dashboard} from './Dashboards';
export {Report} from './Reports';
export {Sharing} from './Sharing';
export {Home, Collection} from './Home';
export {default as Logout} from './Logout';
export {Processes, ProcessReport} from './Processes';
