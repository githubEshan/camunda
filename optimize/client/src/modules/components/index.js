/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
 * one or more contributor license agreements. See the NOTICE file distributed
 * with this work for additional information regarding copyright ownership.
 * Licensed under the Camunda License 1.0. You may not use this file
 * except in compliance with the Camunda License 1.0.
 */

export {Table} from './Table';
export {Modal} from './Modal';
export {Deleter} from './Deleter';
export {Select} from './Select';
export {BPMNDiagram, ClickBehavior, PartHighlight} from './BPMNDiagram';
export {CopyToClipboard} from './CopyToClipboard';
export {ReportRenderer, NoDataNotice} from './ReportRenderer';
export {ErrorBoundary} from './ErrorBoundary';
export {ActionItem} from './ActionItem';
export {ShareEntity} from './ShareEntity';
export {DashboardRenderer, DiagramScrollLock} from './DashboardRenderer';
export {DefinitionSelection, VersionPopover, TenantPopover} from './DefinitionSelection';
export {TargetValueBadge} from './TargetValueBadge';
export {ErrorPage} from './ErrorPage';
export {DatePicker} from './DatePicker';
export {ColorPicker} from './ColorPicker';
export {EntityName} from './EntityName';
export {EntityNameForm} from './EntityNameForm';
export {HeatmapOverlay} from './HeatmapOverlay';
export {EntityList} from './EntityList';
export {UserTypeahead} from './UserTypeahead';
export {DateRangeInput} from './DateRangeInput';
export {LastModifiedInfo} from './LastModifiedInfo';
export {ReportDetails} from './ReportDetails';
export {InstanceCount} from './InstanceCount';
export {DocsLink} from './DocsLink';
export {Tabs} from './Tabs';
export {Checklist} from './Checklist';
export {SelectionPreview} from './SelectionPreview';
export {DownloadButton} from './DownloadButton';
export {BulkDeleter} from './BulkDeleter';
export {AlertModal} from './AlertModal';
export {AlertsDropdown} from './AlertsDropdown';
export {DurationChart} from './DurationChart';
export {ReportTemplateModal, DashboardTemplateModal} from './TemplateModal';
export {KpiCreationModal} from './KpiCreationModal';
export {EmptyState} from './EmptyState';
export {PageTitle} from './PageTitle';
export {TextEditor} from './TextEditor';
export {EntityDescription} from './EntityDescription';
export {Popover} from './Popover';
export {InstanceViewTable} from './InstanceViewTable';
export {TenantInfo} from './TenantInfo';
export {TargetSelection} from './TargetSelection';
export {Loading} from './Loading';
export {Breadcrumbs} from './Breadcrumbs';
