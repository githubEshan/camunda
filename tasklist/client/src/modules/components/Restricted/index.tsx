/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
 * one or more contributor license agreements. See the NOTICE file distributed
 * with this work for additional information regarding copyright ownership.
 * Licensed under the Camunda License 1.0. You may not use this file
 * except in compliance with the Camunda License 1.0.
 */

import {usePermissions} from 'modules/hooks/usePermissions';
import type {Permissions} from 'modules/types';

type Props = {
  children: React.ReactNode;
  scopes: Permissions;
  fallback?: React.ReactNode;
};

const Restricted: React.FC<Props> = ({children, scopes, fallback}) => {
  const {hasPermission} = usePermissions(scopes);

  if (!hasPermission) {
    return <>{fallback}</>;
  }

  return <>{children}</>;
};

export {Restricted};
