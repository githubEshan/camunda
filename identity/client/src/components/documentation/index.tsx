import styled from "styled-components";
import { linkInverse, spacing04 } from "@carbon/elements";
import { Link } from "@carbon/react";
import { FC, ReactNode } from "react";
import useTranslate from "../../utility/localization";
import { docsUrl } from "src/configuration";

export const DocumentationDescription = styled.p`
  margin-top: ${spacing04};
  max-width: none;
  text-align: left;
`;

export const LightLink = styled(Link)`
  display: inline;
  color: ${linkInverse} !important;
`;

type DocumentationLinkProps = {
  children?: ReactNode;
  light?: boolean;
  path?: string;
};

export const documentationHref = (
  docsUrl: string,
  path: DocumentationLinkProps["path"],
) => `${docsUrl}${path}`;

export const DocumentationLink: FC<DocumentationLinkProps> = ({
  path = "",
  light = false,
  children,
}) => {
  const LinkComponent = light ? LightLink : Link;
  const { Translate } = useTranslate();

  return (
    <LinkComponent
      href={documentationHref(docsUrl, path)}
      data-test="documentation-link"
      target="_blank"
    >
      {children || <Translate>documentation</Translate>}
    </LinkComponent>
  );
};
