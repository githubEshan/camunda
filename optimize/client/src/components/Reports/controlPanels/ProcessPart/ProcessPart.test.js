/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
 * one or more contributor license agreements. See the NOTICE file distributed
 * with this work for additional information regarding copyright ownership.
 * Licensed under the Camunda License 1.0. You may not use this file
 * except in compliance with the Camunda License 1.0.
 */

import {shallow} from 'enzyme';

import {BPMNDiagram} from 'components';

import ProcessPart from './ProcessPart';

it('should display a button if no process part is set', () => {
  const node = shallow(<ProcessPart />);

  expect(node.find('Button').filterWhere((n) => n.text() === 'Process instance part')).toExist();
});

it('should not display the button if process part is set', () => {
  const node = shallow(<ProcessPart processPart={{start: 'a', end: 'b'}} />);

  expect(
    node.find('Button').filterWhere((n) => n.text() === 'Process instance part')
  ).not.toExist();
});

it('should show a preview of the process part', () => {
  const node = shallow(
    <ProcessPart processPart={{start: 'a', end: 'b'}} flowNodeNames={{a: 'Start Node'}} />
  );

  expect(node.find('ActionItem')).toMatchSnapshot();
});

it('should remove the process part', () => {
  const spy = jest.fn();
  const node = shallow(<ProcessPart processPart={{start: 'a', end: 'b'}} update={spy} />);

  node.find('ActionItem').at(0).prop('onClick')({stopPropagation: jest.fn()});

  expect(spy).toHaveBeenCalledWith(null);
});

it('should open a modal when clicking the button', () => {
  const node = shallow(<ProcessPart processPart={{start: 'a', end: 'b'}} />);

  node.find('ActionItem').prop('onEdit')();

  expect(node.state('modalOpen')).toBe(true);
});

it('should show the bpmn diagram', () => {
  const node = shallow(<ProcessPart processPart={{start: 'a', end: 'b'}} />);

  node.find('.ProcessPart__current').simulate('click');

  expect(node.find(BPMNDiagram)).toExist();
});

it('should deselect a node when it is clicked and already selected', () => {
  const node = shallow(<ProcessPart />);

  const flowNode = {id: 'nodeId', name: 'Some node'};

  node.setState({
    modalOpen: true,
    start: flowNode,
  });

  node.instance().selectNode(flowNode);

  expect(node.state('start')).toBe(null);
});

it('should display a warning if there is no path between start and end node', () => {
  const node = shallow(<ProcessPart />);

  const flowNode = {id: 'nodeId', name: 'Some node'};

  node.setState({
    modalOpen: true,
    start: flowNode,
    end: {id: 'anId'},
    hasPath: false,
  });

  expect(node.find('InlineNotification').prop('subtitle')).toMatch(
    'Report results may be empty or misleading.'
  );
});
