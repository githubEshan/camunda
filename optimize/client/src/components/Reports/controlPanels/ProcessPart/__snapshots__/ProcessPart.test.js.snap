// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show a preview of the process part 1`] = `
<ActionItem
  onClick={[Function]}
  onEdit={[Function]}
>
  <ForwardRef(Tag)
    size="sm"
    type="blue"
  >
    Process instance part
  </ForwardRef(Tag)>
  <div>
    <b>
      Start Node
    </b>
    <span>
       
      and
       
    </span>
    <b>
      b
    </b>
  </div>
</ActionItem>
`;
